src/MCPClient/lib/clientScripts/convert_dataverse_structure.py

```diff
@@ -0,0 +1,499 @@
+    ext_ = os.path.splitext(fname)[1]
+    if ext_.lower() == '.zip':
+        return True
+    return False
+
+
+def add_ddi_xml(job, sip, json_metadata):
+    """Create a DDI XML data block and add this to the METS."""
+    ddi_root = create_ddi(job, json_metadata)
+    if create_ddi is None:
+        return None
+    sip.add_dmdsec(md=ddi_root, mdtype="DDI")
+    return sip
+
+
+def add_metadata_ref(sip, md_name, md_loc):
+    """Add a single mdref to the METS file."""
+    sip.add_dmdsec(
+        md=md_loc,
+        mdtype="OTHER",
+        mode="mdref",
+        label=md_name,
+        loctype="OTHER",
+        otherloctype="SYSTEM",
+    )
+    return sip
+
+
+def add_md_dir_to_structmap(sip):
+    """Add the metadata directory to the structmap."""
+    md_dir = metsrw.FSEntry(path="metadata", use=None, type="Directory")
+    sip.add_child(md_dir)
+    # Add dataset.json to the fileSec output.
+    f = metsrw.FSEntry(
+        path="metadata/dataset.json", use="metadata",
+        file_uuid=str(uuid.uuid4())
+    )
+    # Add dataset.json to the metadata fileSec group.
+    md_dir.add_child(f)
+    return sip
+
+
+def add_dataset_files_to_md(job, sip, latest_version_md, contact_information):
+    # Add original files to the METS document.
+    files = {}
+    files = latest_version_md.get('files')
+    if not files:
+        return None
+
+    # Signal to users the existence of zip files in this transfer.
+    zipped_file = False
+
+    # Signal to users that this transfer might consist of metadata only.
+    if len(files) is 0:
+        logger.info(
+            "Metadata only transfer? There are no file entries in this "
+            "transfer's metadata.")
+
+    for file_json in files:
+        is_restricted = file_json.get("restricted")
+        if is_restricted is True and contact_information:
+            logger.error(
+                "Restricted dataset files may not have transferred "
+                "correctly: %s", contact_information)
+        if file_json.get("dataFile").get("filename").endswith(".tab"):
+            # A Tabular Data File from Dataverse will consist of an original
+            # tabular format submitted by the researcher plus multiple
+            # different representations. We need to map that here.
+            bundle = create_bundle(job, file_json)
+            if bundle:
+                sip.add_child(bundle)
+            else:
+                logger.error(
+                    "Create Dataverse transfer METS failed. "
+                    "Bundle returned: %s", bundle)
+                return None
+        else:
+            path_ = None
+            datafile = file_json.get("dataFile")
+            if datafile:
+                path_ = datafile.get("filename")
+            if path_:
+                if test_if_zip_in_name(path_):
+                    # provide some additional logging around the contents of the
+                    # dataset we're processing.
+                    if not zipped_file:
```

Comment:
That's right. We experienced a few issues with zips and while I'm not sure those issues are still the same now that we're handling bundles better, I wanted  to make sure we log enough about the characteristics of a download in the UI so that there is plenty for the user to debug for themselves if necessary. A breadcrumb if you will.